src/main/scala/sparkz/core/network/NetworkController.scala

```diff
@@ -546,6 +550,17 @@ class NetworkController(settings: NetworkSettings,
     }
   }
 
+  private def closeConnectionFromNode(peerAddress: InetSocketAddress): Unit = {
+    connections = connections.filterNot {
+      case (address, connectedPeer) => address == peerAddress match {
+        case true =>
+          connectedPeer.handlerRef ! CloseConnection
+          context.system.eventStream.publish(DisconnectedPeer(peerAddress))
+          true // exclude the entry from the filtered map
+      }
```

Comment:
`case false` is missing here, so in case of address != peerAddress a scala.MatchError will be thrown.